### DISCUSSION TITLE
how to append data from different runs to a master file

### DISCUSSION THREAD

**Discussion by chincxxx** [jspsych/jsPsych#3394] [Q&A]
Hi all, 

I've been really confused about how to save data trial-by-trial, that is, to append every key trial from different run to a master file that stores all participants responses. I only need it to be saved locally so I tried localSave (), but somehow it only takes the first 2 arguments but not the third argument append as described [here](https://github.com/Tuuleh/jsPsychTutorialProgress/blob/master/jsPsych/docs/markdown_docs/core_library/jspsych-data.md). 

Really need help! Thanks in advance!

**Comment by ChristopheBossens** [comment]
Hi chincxxx,

The append parameter desribed in the localSave() function is used to append additional information to the existing data structure of your experiment and is not meant to append data to a local file (it would be a huge security risk if a script running in your web browser can just append to file on your local file system). 

Is there a specific reason why you want to append to a master file during the experiment? You could also save the data of each individual participant and then have a script that merges these individual files after the experiment. If you really need to do this for each individual trial you could probably use localSave to store the individual trials for each participant as well and then merge all the trials into the master file with a script. 

If you are familiar with Python this is very straightforward to do, let me know if you need help with this.
